spec/system/admin/order_cycles/complex_editing_multiple_updation_spec.rb

```diff
@@ -0,0 +1,286 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+describe '
+    As an administrator
+    I want to see alert for unsaved changes on order cycle edit page
+' do
+  include AuthenticationHelper
+  include WebHelper
+
+  before do
+    Capybara.current_driver = :selenium_chrome_headless
+    Capybara.javascript_driver = :selenium_chrome_headless
+    Capybara.default_max_wait_time = 10
+  end
+
+  after do
+    Capybara.use_default_driver
+  end
+
+  it "Alerts for unsaved changes on general settings (/edit) page" do
+    oc = create(:order_cycle)
+    login_as_admin_and_visit edit_admin_order_cycle_path(oc)
+
+    # Wait for the page to finish loading
+    sleep 3
+
+    # Expect correct values
+    expect(page.find('#order_cycle_name').value).to eq(oc.name)
+    expect(page).to have_content "COORDINATOR #{oc.coordinator.name}"
+    expect(page).to have_button('Save', disabled: true)
+    expect(page).to have_button('Save and Next', disabled: true)
+
+    # First change
+    fill_in 'order_cycle_name', with: 'Bonnie'
+    fill_in 'order_cycle_orders_open_at', with: '2020-01-06 06:00:00 +0000'
+    fill_in 'order_cycle_orders_close_at', with: '2020-01-07 06:00:00 +0000'
+    expect(page).to have_content('You have unsaved changes')
+    expect(page).to have_button('Save', disabled: false)
+    expect(page).to have_button('Save and Next', disabled: false)
+    expect(page).to have_button('Cancel', disabled: false)
+    expect(page).to have_button('Next', disabled: true)
+
+    # Trying to go to another page with unsaved changes
+    click_link 'Orders'
+
+    # expect an alert about unsaved changes
+    expect(page.driver.browser.switch_to.alert.text)
```

Comment:
Ok, now I understand why you wanted to include Selenium.

The way we do this with capybara/cuprite (ie. our current stack, without adding any Gems or changing any configuration) is: 

```ruby
    dismiss_confirm "" do
      click_link 'Orders'
    end
```
instead of:

```ruby
    click_link 'Orders'
    expect(page.driver.browser.switch_to.alert.text)
    page.driver.browser.switch_to.alert.dismiss
```